App.tsx

```diff
@@ -4,13 +4,19 @@ import DiabetesApp from './src/DiabetesApp';
 
 // Providers
 import { AuthProvider } from './src/context/FirebaseContext';
+import { ErrorProvider } from './src/context/ErrorContext';
+import { Provider as PaperProvider } from 'react-native-paper';
 
 const App = () => {
   return (
-    <AuthProvider>
-      <DiabetesApp />
-    </AuthProvider>
+    <PaperProvider>
```

Comment:
Added the `PaperProvider` since its required to use it to utilize the react native paper `Modal` component